components/camel-olingo4/camel-olingo4-component/src/main/java/org/apache/camel/component/olingo4/Olingo4Consumer.java

```diff
@@ -82,7 +83,15 @@ public void onCanceled() {
                 throw error[0];
             }
 
-            return ApiConsumerHelper.getResultsProcessed(this, result[0], isSplitResult());
+            //
+            // Allow consumer idle properties to properly handle an empty polling response
+            //
+            int processed = ApiConsumerHelper.getResultsProcessed(this, result[0], isSplitResult());
+            if (result[0] instanceof ClientEntitySet && (((ClientEntitySet) result[0]).getEntities().isEmpty())) {
```

Comment:
Maybe put this code into `ApiConsumerHelper` as a new method with some kind of name, isEmptyResult or something?

Reply:
Unfortunately, that's not possible since that would pollute ApiConsumerHelper with a dependency on Olingo4 classes, ie. ClientEntitySet. By keeping it here it confines such dependencies.